flexmeasures/data/models/reporting/__init__.py

```diff
@@ -75,7 +75,7 @@ def fetch_data(
             )
 
             # store data source as local variable
-            for source in bdf.sources.unique():
+            for source in bdf.sources.unique().dropna():
```

Comment:
`bdf.lineage.sources` might also work.

https://github.com/SeitaBV/timely-beliefs/blob/main/timely_beliefs/beliefs/__init__.py#L109

I wouldn't have expected any NaN sources, actually. How come there are any?

Reply:
Using `bdf.lineage.sources`  also yields a NaN source. 

The following code snippet should be useful to reproduce this behavior using the data from 2023.

```python
from datetime import datetime, timedelta
import pytz

from flexmeasures.data.models.time_series import Sensor

sensor = Sensor.query.get(6)

timezone = pytz.timezone("Europe/Amsterdam")
event_starts_after = tz.localize(datetime(2023, 1, 2))
event_ends_before = tz.localize(datetime(2023, 1, 23))
resolution = timedelta(seconds=3600)

bdf = sensor.search_beliefs(
    event_starts_after=event_starts_after,
    event_ends_before=event_ends_before,
    resolution=resolution,
    beliefs_before=belief_time,
    **_tb_query,
)
```

```python
>> bdf.sources.isna().sum()
37
```